kinto/core/resource/schema.py

```diff
@@ -150,3 +151,71 @@ class BookmarkSchema(ResourceSchema):
 
     def preparer(self, appstruct):
         return strip_whitespace(appstruct)
+
+
+class CSVQuerystring(colander.SchemaNode):
+    """String field representing a list of attributes."""
+
+    schema_type = colander.String
+    missing = colander.drop
+    error_message = "The value should be a list of comma separated field names"
+
+    def deserialize(self, cstruct=colander.null):
+        params = super(CSVQuerystring, self).deserialize(cstruct)
+        if params is colander.drop:
+            return params
+        else:
+            return params.split(',')
+
+
+class HeaderQuotedInteger(colander.SchemaNode):
+    """Integer between "" used in precondition headers."""
+
+    schema_type = colander.String
+    missing = colander.drop
+    error_message = "The value should be integer between double quotes"
+    validator = colander.Any(colander.Regex('^"([0-9]+?)"$', msg=error_message),
+                             colander.Regex('\*'))
+
+    def deserialize(self, cstruct=colander.null):
+        if isinstance(cstruct, six.string_types):
+            try:
+                cstruct = decode_header(cstruct)
+            except:
+                raise colander.Invalid(self, msg=self.error_message)
+
+        param = super(HeaderQuotedInteger, self).deserialize(cstruct)
+        if param is colander.drop or param == '*':
+            return param
+        else:
+            return int(param[1:-1])
+
+
+class HeaderSchema(colander.MappingSchema):
+    if_match = HeaderQuotedInteger(name='If-Match', missing=colander.drop)
+    if_none_match = HeaderQuotedInteger(name='If-None-Match', missing=colander.drop)
+    response_behaviour = colander.SchemaNode(colander.String(),
+                                             name='Response-Behavior',
+                                             validator=colander.OneOf(
+                                                 ['full', 'light', 'diff']),
+                                             missing=colander.drop)
```

Comment:
Is `decode_header` not necessary for this one? Maybe we could run `decode_header` here before deserailize each sub-node so that we have it one place only? Or do it in Cornice instead...